### PR TITLE
[reshare] Add per-peer metrics for latest share and acknowledgements

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds two new `Family` metrics to the DKG actor to track the latest shares and acknowledgements received per-peer.

### CODE REVIEW

File: examples/reshare/src/dkg/actor.rs [hunk 1]
```diff
@@ -141,6 +159,18 @@ where
         context.register("failed_epochs", "failed epochs", failed_epochs.clone());
         context.register("our_reveals", "our share was revealed", our_reveals.clone());
         context.register("all_reveals", "all share reveals", all_reveals.clone());
+        let latest_share = Family::<Peer, Gauge>::default();
```
Comment (patrick-ogrady) [inline examples/reshare/src/dkg/actor.rs]: move this definition up to the block of Counters (for consistency)

File: examples/reshare/src/dkg/state.rs [hunk 2]
```diff
@@ -526,9 +526,9 @@ impl<V: Variant, C: Signer> Dealer<V, C> {
         epoch: EpochNum,
         player: C::PublicKey,
         ack: PlayerAck<C::PublicKey>,
-    ) {
+    ) -> bool {
```
Comment (patrick-ogrady) [inline examples/reshare/src/dkg/state.rs:529]: Can we add a unit test for this?

File: examples/reshare/src/dkg/actor.rs [hunk 3]
```diff
@@ -376,7 +413,15 @@ where
...
-                                        ds.handle(&mut storage, epoch, sender_pk, ack).await;
+                                        let valid = ds
+                                            .handle(&mut storage, epoch, sender_pk.clone(), ack)
+                                            .await;
+                                        if valid {
+                                            let _ = self
+                                                .latest_ack
+                                                .get_or_create(&Peer::new(&sender_pk))
+                                                .try_set_max(epoch.get());
+                                        }
```
[the remark this answers is not in the document]
Comment (clabby) [reply examples/reshare/src/dkg/actor.rs:424]: This is fine.

File: examples/reshare/src/dkg/state.rs [hunk 4]
```diff
@@ -654,3 +654,169 @@ impl<V: Variant, C: Signer> Player<V, C> {
...
+
+            assert!(
+                !result,
+                "handle should return false when crypto dealer is None"
+            );
+        });
+    }
+
+    #[test_traced]
+    fn test_dealer_handle_returns_true_for_valid_ack() {
```
Comment (patrick-ogrady) [inline examples/reshare/src/dkg/state.rs:783]: Can you add a test for a duplicate?

File: examples/reshare/src/dkg/actor.rs [hunk 5]
```diff
@@ -376,7 +413,15 @@ where
                                 }
                                 Message::Ack(ack) => {
                                     if let Some(ref mut ds) = dealer_state {
-                                        ds.handle(&mut storage, epoch, sender_pk, ack).await;
+                                        let valid = ds
```
Comment (patrick-ogrady) [inline examples/reshare/src/dkg/actor.rs]: nit: `let added`
